sql/row.go

```diff
@@ -89,6 +89,17 @@ type RowIter interface {
 	Closer
 }
 
+// ExtendedRowIter is an iterator that produces rows. It is an extended interface over RowIter. This interface may
+// change over time.
+type ExtendedRowIter interface {
```

Comment:
I would leave this in the plan package for now, seems very specific to the implementation of stored procedures